### PR TITLE
Update SstFileWriter fuzzer to iterate and check all key-value pairs

### PR DESCRIPTION
**Pull request by cheng-chang**
as title

Test Plan:
cd fuzz && make sst_file_writer_fuzzer && ./sst_file_writer_fuzzer

### CODE REVIEW

File: fuzz/proto/db_operation.proto [hunk 1]
```diff
@@ -10,7 +10,6 @@ syntax = "proto2";
 
 enum OpType {
   PUT = 0;
-  MERGE = 1;
```
Comment (siying) [inline fuzz/proto/db_operation.proto:13]: Can you explain why it is removed somewhere?

File: fuzz/sst_file_writer_fuzzer.cc [hunk 2]
```diff
@@ -62,27 +91,31 @@ DEFINE_PROTO_FUZZER(DBOperations& input) {
   rocksdb::SstFileWriter writer(env_options, options);
   rocksdb::Status s = writer.Open(sstfile);
   CHECK_OK(s);
+  std::map<std::string, std::string> kv;
```
Comment (siying) [inline fuzz/sst_file_writer_fuzzer.cc:94]: SstFileWriter doesn't sort data, so it mirrors to a vector. Why map?

File: fuzz/sst_file_writer_fuzzer.cc [hunk 3]
```diff
@@ -62,27 +91,31 @@ DEFINE_PROTO_FUZZER(DBOperations& input) {
...
-      case OpType::MERGE:
-        s = writer.Merge(op.key(), op.value());
-        CHECK_OK(s);
-        break;
-      case OpType::DELETE:
+      }
+      case OpType::DELETE: {
         s = writer.Delete(op.key());
         CHECK_OK(s);
+        kv.erase(op.key());
```
Comment (siying) [inline fuzz/sst_file_writer_fuzzer.cc:106]: I thought SstFileReader just return each record put, delete or merge. Does it apply delete?

File: fuzz/sst_file_writer_fuzzer.cc (hunk 2 above)
Comment (siying) [reply fuzz/sst_file_writer_fuzzer.cc:94]: Never mind. I was wrong.

In that case, maybe SstFileReader isn't the best interface to do the validation. We need something lower level. Something like directly returning from TableReader.

File: fuzz/proto/db_operation.proto (hunk 1 above)
Comment (cheng-chang) [reply fuzz/proto/db_operation.proto:13]: just want to keep it simple to start with simple operations like put, delete.

File: fuzz/sst_file_writer_fuzzer.cc (hunk 2 above)
Comment (pdillinger) [reply fuzz/sst_file_writer_fuzzer.cc:94]: I agree. I would rather see a "round trip" of the data where you can compare it with `input.operations()` again; no need for a 'kv' map or vector.

**Comment by cheng-chang** [comment]
@siying @pdillinger thanks for the suggestion, PTAL.

**Review by pdillinger — APPROVED** [review]
LGTM

**Pull request closed without merging**